enhancements/olm-admission-webhooks.md

```diff
@@ -114,9 +116,37 @@ The certificate creation and lifecycle management is planned to utilize the same
 
 There is a great deal of community support for integration with certificate management projects such as the [service-ca operator](https://github.com/openshift/service-ca-operator) or [cert-manager](https://github.com/jetstack/cert-manager). Cluster Admins should be given the opportunity to integrate with either of these tools be disabling the OLM management of certs and adding appropriate annotations to the deployment and webhook resources via the [SubscriptionConfig resource](https://github.com/operator-framework/operator-lifecycle-manager/blob/8985872bfd5888c8d7d49e4dfc9be162a87691e7/pkg/api/apis/operators/v1alpha1/subscription_types.go#L41-L87).
 
+### Deployment Installation Order
+
+There are a number of customer usecases in which the rollout order of deployments matters. Consider the following scenarios:
+
+- An operator may be written to assume that any CR it reconciles has been acted upon by a webhook defined in its CSV. In this situation, OLM should provide the Operator Author with the ability to install the admission webhook before the operator.
+- An operator may be written such that it expects to perform some action prior to starting the webhook. This logic may involve creating a CR that has special values that would be changed by the webhook defined in the CSV. In this situation, OLM should provide the Operator Author with the ability to install the operator prior to the webhook.
+
+Both of the usecases described above can be addressed by updating OLM to:
+
+- Create Deployments defined in the CSV in order.
+- When creating the deployments defined in the CSV:
+  - Waiting for the Deployment to reach the `READY` state.
+  - Checking if the deployment owns any webhook or APIServices resources and if so, create the required resource.
+
+### Webhook Upgrades
+
+OLM must support the addition, subtraction, and upgrade of webhooks between versions of a CSV:
+
+- If an admission webhook is removed in a new version of the CSV, the on cluster Webhook should be deleted.
+- If a new webhook is added in a new version of the CSV, the webhook should be deployed.
+- If the webhook is updated in a new version of the CSV, the existing webhook should be disabled before upgrading the operator.
+
 ### Risks and Mitigations
 
-- Webhooks allow operators to intercept core resource and could wreak havok on clusters if incorrectly configured. As such, users should be notified when an operator bundle includes a webhook.
+- Webhooks allow operators to intercept core resource and could wreak havoc on clusters if incorrectly configured. As such, users should be notified when an operator bundle includes a webhook.
```

Comment:
nit: does "notify" mean that `InstallPlans` are switched to manual when a webhook is detected?

Reply:
I had imagined that there might be a message in the frontend or something similar.

Reply:
I think this makes sense but should be outside of the scope for now to ensure we can deliver this. In the future I think we should continue to make the webhooks something that surfaces in the `InstallPlan` and `Install`.  We can classify the webhooks that touch objects of certain API groups (that represent core features) and annotate them as extra privileged. A UI can take this up and surface it to the user differently than web hooks without that annotation.

Reply:
/sgtm